cypress/integration/diagrams.spec.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+describe('renders diagram', () => {
+  beforeEach(() => {
+    cy.visitTestEditor()
+  })
+
+  it('markmap', () => {
+    cy.codemirrorFill('```markmap\n- pro\n- contra\n```')
+    cy.getMarkdownBody()
+      .find('svg')
+      .should('be.visible')
+  })
+
+  it('vega-lite', () => {
+    cy.codemirrorFill('```vega-lite\n{"$schema":"https://vega.github.io/schema/vega-lite/v4.json","data":{"values":[{"a":"","b":28}]},"mark":"bar","encoding":{"x":{"field":"a"},"y":{"field":"b"}}}\n```')
+    cy.getMarkdownBody()
+      .find('div.vega-embed canvas')
+      .should('be.visible')
+  })
+
+  it('graphviz', () => {
+    cy.codemirrorFill('```graphviz\ngraph {\na -- b\n}\n```')
+    cy.getMarkdownBody()
+      .find('svg')
+      .should('be.visible')
+  })
+
+  it('mermaid', () => {
+    cy.codemirrorFill('```mermaid\ngraph TD;\n    A-->B;\n```')
+    cy.getMarkdownBody()
+      .find('div.mermaid > svg')
```

Comment:
div

Reply:
Same as above

Reply:
I think Tilman is suggesting to use `.mermaid > svg` or just `.mermaid` for the test. I guess if one wants the svg depends on how the error case looks. If in case of an error the output would also contain some HTML Tag with the class `.mermaid`, but not an `svg` this should be included…

Reply:
Then you should just use the class or use a data-cy tag. We're testing the replacer, not mermaid.